.github/workflows/complete.yml

```diff
@@ -156,3 +156,25 @@ jobs:
       - name: copy to gs
         run: gsutil cp ./spark/ingestion/target/feast-ingestion-spark-${GITHUB_SHA}.jar gs://feast-jobs/spark/ingestion/
 
+  test-end-to-end:
+    runs-on: [self-hosted]
```

Comment:
The self-hosted runners don't clean up. If you aren't using docker then all the state will remain. I dont think this test is safe, it's going to fall over for our current runners because they are already so unstable.

Reply:
I don't use docker. It's regular test, which github runner was designed for, hence there should be way to easily clean up. Right now for example python w/ environment (and installed packages) being cleaned up by python action

Reply:
Also, we're on preemptible nodes right now. So we have additional clean up

Reply:
> I don't use docker. It's regular test, which github runner was designed for, hence there should be way to easily clean up. Right now for example python w/ environment (and installed packages) being cleaned up by python action

ok, but self-hosted runners don't clean up right? state just hangs around as far as I know.